src/org/zaproxy/zap/extension/frontEndScanner/ExtensionFrontEndScanner.java

```diff
@@ -0,0 +1,358 @@
+
+import java.awt.CardLayout;
+import java.awt.Font;
+import java.lang.Exception;
+import java.lang.String;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.UUID;
+
+import javax.swing.ImageIcon;
+import javax.swing.JTextPane;
+
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.OutputDocument;
+import net.htmlparser.jericho.Source;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.proxy.ProxyListener;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.history.ProxyListenerLog;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.api.API;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptType;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+
+/**
+ * A ZAP extension which allow to run scripts in the browser to detect
+ * vulnerabilities in web applications relying heavily on Javascript.
+ */
+public class ExtensionFrontEndScanner extends ExtensionAdaptor implements ProxyListener {
+
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionFrontEndScanner";
+
+    private static final String SCRIPT_TYPE_CLIENT_ACTIVE = "client-side-active";
+    private static final String SCRIPT_TYPE_CLIENT_PASSIVE = "client-side-passive";
+
+    // The i18n prefix, by default the package name - defined in one place to make it easier
+    // to copy and change this example
+    protected static final String PREFIX = "frontEndScanner";
+
+    private static final String RESOURCE = "/org/zaproxy/zap/extension/frontEndScanner/resources";
+    private static final String FRONT_END_SCANNER = Constant.getZapHome() + "/frontEndScanner/front-end-scanner.js";
+    private static final String SCRIPTS_FOLDER = Constant.getZapHome() + "/scripts/scripts/";
+
+    private ZapMenuItem menuFrontEndScanner;
+
+    private boolean frontEndScannerEnabled = true;
+
+    private ScriptType activeScriptType;
+    private ScriptType passiveScriptType;
+    private ExtensionScript extensionScript;
+
+    private FrontEndScannerAPI api;
+
+    private static final Logger LOGGER = Logger.getLogger(ExtensionFrontEndScanner.class);
+
+    private static final List<Class<? extends Extension>> DEPENDENCIES;

```

Comment:
Not currently used

Reply:
But should be.
Added the `getDependencies()` declaration.